src/main.js

```diff
@@ -336,19 +337,24 @@ function configureCrashReporter() {
 		// need to change that directory to the provided one
 		console.log(`Found --crash-reporter-directory argument. Setting crashDumps directory to be '${crashReporterDirectory}'`);
 		app.setPath('crashDumps', crashReporterDirectory);
+
+		enableCrashReporter = true;
 	}
 
-	// Otherwise we configure the crash reporter from product.json
+	// Otherwise we configure the crash reporter from product.json if enabled
 	else {
 		const appCenter = product.appCenter;
-		if (appCenter) {
-			const isWindows = (process.platform === 'win32');
-			const isLinux = (process.platform === 'linux');
-			const isDarwin = (process.platform === 'darwin');
-			const crashReporterId = argvConfig['crash-reporter-id'];
-			const uuidPattern = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/i;
-			if (uuidPattern.test(crashReporterId)) {
-				if (isWindows) {
+		const crashReporterId = argvConfig['crash-reporter-id'];
+		const uuidPattern = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/i;
+		if (
+			appCenter &&							// only when appCenter is configured
+			product.quality === 'insider' &&		// only for insiders
+			typeof crashReporterId === 'string' &&	// only if we have a crash reporter ID
+			uuidPattern.test(crashReporterId) &&	// only if the crash reporter ID is valid
+			crashReporterId.endsWith('0')			// only if the crash reporter ID ends with `0`
```

Comment:
This is a heuristic I picked, but maybe you have a better idea.

Reply:
@deepak1556 I ran some experiments generating `30000` UUIDs to roughly simulate our number of present insider users and typically end up with a number around `~1900` that match. Would you think that number if sufficient to find crashes?